### PR TITLE
Speed up of plot_discrete_cdfs by 2 orders of magnitude

### LINKED ISSUE #298: `test_plot_cdfs` test is extremely slow
This test takes extremely long, in the order of minutes. I wouldn't be surprised if it's over half of all testing time.

https://github.com/quaquel/EMAworkbench/blob/b76b487afc5fbaaa34e5a055269657ff15fa908e/test/test_analysis/test_regional_sa.py#L14-L23

### PR DESCRIPTION
**Pull request by quaquel**
This pull request does 3 things

1. adds a regional sensitivity analysis example
2. speed up of `plot_discrete_cdfs` by 2 orders of magnitude
3. bugfix in `plot_cdfs` for handling scenario_id, policy, and model

Closes #298

### CODE REVIEW

**Comment by EwoutH** [comment]
Thanks, will review later this week.

Is this in any way breaking for current users / workflows?

**Comment by quaquel** [comment]
It is hard to assess how large the impact will be, but I am going through the examples right now and about half of them have to be changed because of this. I will also have to replace all datafiles used in testing and for the visualization examples.

**Comment by quaquel** [comment]
> Thanks, will review later this week.
> 
> Is this in any way breaking for current users / workflows?

this change won't break anything major because it is just a visual change.

File: ema_workbench/analysis/regional_sa.py [hunk 1]
```diff
@@ -82,46 +82,48 @@ def plot_discrete_cdf(ax, unc, x, y, xticklabels_on, ccdf):
...
-            ax.scatter(
-                x_plot[1],
-                y_plot[0],
-                edgecolors=cp[i + 1],
-                facecolors="white",
-                linewidths=1,
-                zorder=2,
-            )
+            ax.plot(x_plot, y_plot, c=cp[i + 1], label=i == 1, marker="o")
+            # ax.scatter(
```
Comment (EwoutH) [inline ema_workbench/analysis/regional_sa.py:86]: I would say just delete it.

File: ema_workbench/analysis/regional_sa.py [hunk 2]
```diff
@@ -82,46 +82,48 @@ def plot_discrete_cdf(ax, unc, x, y, xticklabels_on, ccdf):
...
-                x_plot[1],
-                cum_freq_un,
-                edgecolors="darkgrey",
-                facecolors="white",
-                linewidths=1,
-                zorder=1,
+            ax.plot(
+                x_plot, [cum_freq_un] * 2, lw=1, c="darkgrey", zorder=1, label="x==y", marker="o"
             )
+            # ax.scatter(
```
Comment (EwoutH) [inline ema_workbench/analysis/regional_sa.py:111]: Also delete

File: ema_workbench/analysis/regional_sa.py [hunk 3]
```diff
@@ -256,10 +258,18 @@ def plot_cdfs(x, y, ccdf=False):
     x = x.copy()
 
     try:
-        x = x.drop("scenario", axis=1)
+        x = x.drop("scenario_id", axis=1)
     except KeyError:
```
Comment (EwoutH) [inline ema_workbench/analysis/regional_sa.py:262]: Do we need to add an explicit warning here?

File: ema_workbench/analysis/regional_sa.py [hunk 4]
```diff
@@ -256,10 +258,18 @@ def plot_cdfs(x, y, ccdf=False):
...
         pass
 
+    for entry in ["model", "policy"]:
+        if x.loc[:, entry].unique().shape != (1,):
+            continue
+        try:
+            x = x.drop(entry, axis=1)
+        except KeyError:
```
Comment (EwoutH) [inline ema_workbench/analysis/regional_sa.py:270]: Explicit warning needed?

File: ema_workbench/examples/regional_sa_flu.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+from ema_workbench import ema_logging, load_results
+
+fn = "./data/1000 flu cases with policies.tar.gz"
+results = load_results(fn)
+x, outcomes = results
+
+y = outcomes["deceased population region 1"][:, -1] > 1000000
+
+fig = regional_sa.plot_cdfs(x, y)
+# regional_sa.plot_cdfs(x, y, ccdf=True)
```
Comment (EwoutH) [inline ema_workbench/examples/regional_sa_flu.py:17]: Why are they commented out? Maybe add comment explaining that.

File: ema_workbench/analysis/regional_sa.py (hunk 3 above)
Comment (quaquel) [reply ema_workbench/analysis/regional_sa.py:262]: No, scenario_id, model, and policy are workbench defaults. to keep the code a bit more generic we remove these if they are available (and default to 1 category for model and policy).

File: ema_workbench/analysis/regional_sa.py (hunk 4 above)
Comment (quaquel) [reply ema_workbench/analysis/regional_sa.py:270]: No, scenario_id, model, and policy are workbench defaults. to keep the code a bit more generic we remove these if they are available (and default to 1 category for model and policy).

File: ema_workbench/examples/regional_sa_flu.py (hunk 5 above)
Comment (quaquel) [reply ema_workbench/examples/regional_sa_flu.py:17]: just copy paste from test code. Will remove them.

File: ema_workbench/analysis/regional_sa.py (hunk 1 above)
Comment (quaquel) [reply ema_workbench/analysis/regional_sa.py:86]: agree

File: ema_workbench/analysis/regional_sa.py (hunk 2 above)
Comment (quaquel) [reply ema_workbench/analysis/regional_sa.py:111]: agree

**Comment by EwoutH** [comment]
Before this PR:
278872579-4a3ac6d3-b99c-4f42-a1f7-e3b4fbc6e3e9
After this PR:
Screenshot_209

Goes from 345 seconds to 2.

And looking at the CI test times, it goes from 8 to 10 minutes to around 3 minutes.

Great work!

**Review by EwoutH — APPROVED** [review]

**Comment by EwoutH** [comment]
Could you make the PR title more descriptive (for the changelog) and then feel free to (squash and) merge.

**Pull request merged**